insights/client/schedule.py

```diff
@@ -13,27 +13,28 @@
 
 
 class InsightsSchedule(object):
+
     """
     Set the cron schedule
     """
     def __init__(self, set_cron=True):
-        if set_cron and not self.already_linked():
-            self.set_daily()
+        if set_cron and not self.already_linked(CRON_WEEKLY + APP_NAME) and not self.already_linked(CRON_DAILY + APP_NAME):
+            self.set_daily(CRON_DAILY + APP_NAME)
 
-    def already_linked(self):
+    def already_linked(self, cronfile):
         """
         Determine if we are already scheduled
         """
-        if os.path.isfile(CRON_WEEKLY + APP_NAME):
+        if os.path.isfile(cronfile):
             logger.debug('Found cron.weekly')
             return True
-        elif os.path.isfile(CRON_DAILY + APP_NAME):
+        elif os.path.isfile(cronfile):
```

Comment:
You will no longer need this second 'elif' check as it does the same thing as the first. I would remove this one and change the verbiage to 'Found %s' or something along those lines.